spec/shared_examples/checkout.rb

```diff
@@ -0,0 +1,231 @@
+
+    aggregate_failures do
+      expect(page).to have_css('.panel-heading', text: 'Payment Information')
+      expect(name_on_card).to eq full_name
+      expect(page).to have_css('#cvv_link', text: "What's This?")
+    end
+  end
+
+  it 'can save credit card' do
+    fill_in_cc(credit_card)
+    click_button 'Save and Continue'
+
+    expect(page).to have_css('.active', text: 'Confirm')
+  end
+
+  it 'can pay via check' do
+    choose('order_payments_attributes__payment_method_id_2')
+
+    click_button 'Save and Continue'
+
+    expect(page).to have_css('.alert-notice', text: 'Your order has been processed successfully')
+  end
+
+  it 'can add promocode' do
+    fill_in_cc(credit_card)
+    fill_in 'order_coupon_code', with: 'segment'
+
+    click_button 'Save and Continue'
+
+    expect(page).to have_css('.total', text: 'Promotion (Segment)')
+  end
+end
+
+shared_examples 'Confirmation step' do
+  before { save_payment(credit_card, 'UPS Two Day (USD)', address) }
+
+  it 'info is visable' do
+    aggregate_failures do
+      expect(page).to have_css('div[data-hook="checkout_header"]')
+      expect(page).to have_css('.col-sm-3', text: 'Checkout')
+      expect(page).to have_css('.completed', text: 'Address')
+      expect(page).to have_css('.completed', text: 'Delivery')
+      expect(page).to have_css('.completed', text: 'Payment')
+      expect(page).to have_css('.completed', text: 'Address')
+      expect(page).to have_css('.active', text: 'Confirm')
+      expect(page).to have_css('.last', text: 'Complete')
+
+      expect(page).to have_css('.panel-heading', text: 'Confirm')
+      expect(page).to have_css('.col-xs-6', text: 'Billing Address')
+      expect(page).to have_css('.col-xs-6', text: 'Shipping Address')
+      expect(page).to have_css('.col-xs-6', text: 'Shipments')
+      expect(page).to have_css('.col-xs-6', text: 'Payment Information')
+
+      expect(page).to have_css('.active', text: 'Item')
+      expect(page).to have_css('.active', text: 'Price')
+      expect(page).to have_css('.active', text: 'Qty')
+      expect(page).to have_css('.active', text: 'Total')
+      expect(page).to have_css('#line-items', text: 'Ruby on Rails Bag')
+      expect(page).to have_css('#line-items', text: 'Ruby on Rails Tote')
+
+      expect(page).to have_css('.total', text: 'Subtotal')
+      expect(page).to have_css('.total', text: 'Shipping')
+      expect(page).to have_css('.total', text: 'Tax')
+      expect(page).to have_css('.total', text: 'Order Total')
+
+      expect(page).to have_button('Place Order')
+    end
+  end
+
+  it 'can place the order' do
+    click_button 'Place Order'
+
+    aggregate_failures do
+      expect(page).to have_css('.alert-notice', text: 'Your order has been processed successfully')
+      expect(page).to have_css('.button', text: 'Back to Store')
+    end
+  end
+end
+
+shared_examples 'can edit on Confirmation step' do |name, title, order_number = nil|
```

Comment:
split this `shared_example` into small ones (put more test cases inside this `shared_example`), as there is a lot of confusion here